client/gui/CGuiHandler.cpp

```diff
@@ -609,30 +609,31 @@ void CFramerateManager::init()
 
 void CFramerateManager::framerateDelay()
 {
+	accumulatedFrames++;
+
 	ui32 currentTicks = SDL_GetTicks();
-	timeElapsed = currentTicks - lastticks;
+	ui32 timeElapsed = currentTicks - lastticks;
 
 	// FPS is higher than it should be, then wait some time
-	if (timeElapsed < rateticks)
+	if(timeElapsed < rateticks)
 	{
 		SDL_Delay((Uint32)ceil(this->rateticks) - timeElapsed);
 	}
 
+	currentTicks = SDL_GetTicks();
+	// recalculate timeElapsed for external calls via getElapsed()
+	// limit it to 1000 ms to avoid breaking animation in case of huge lag (e.g. triggered breakpoint)
+	timeElapsed = std::min<ui32>(currentTicks - lastticks, 1000);
+
+	lastticks = SDL_GetTicks();
+
 	accumulatedTime += timeElapsed;

```

Comment:
maybe std::max(1, timeElapsed) could be the good way to avoid any issues

Reply:
timeElapsed can not be 0 because now we count SDL_Delay

Reply:
So it will be not less than rateticks